src/Core/Framework/Struct/AssignArrayTrait.php

```diff
@@ -31,4 +39,108 @@ public function assign(array $options)
 
         return $this;
     }
+
+    /**
+     * @param array<array-key, mixed> $options
+     */
+    private function assignRecursive(array $options): self
```

Comment:
what about making this a public method and add this method to the interface instead? :thinking: